File: day_one/part_one/trebuchet.py
# Trebuchet: Sums the first and last digit of each line.
# Advent of Code 2023 - Day 1 - Part 1


def trebuchet_combine(data):
    data1 = data[0]
    data2 = data[-1]
    data = str(data1) + str(data2)

    return int(data)


def trebuchet(file):
    trebuchet_array = []

    with open(file, "r") as file:
        for line in file:
            data = "".join(filter(lambda x: x.isdigit(), line))
            trebuchet_array.append(trebuchet_combine(data))

    return trebuchet_array


result = sum(trebuchet("data.txt"))
print(f"🥞 The result is {result}")

File: day_one/part_one/Dart/trebuchet.dart
// Trebuchet: Sums the first and last digit of each line.
// Advent of Code 2023 - Day 1 - Part 1

import 'dart:convert';
import 'dart:io';

int trebuchetCombine(String data) {
  final data1 = data[0];
  final data2 = data[data.length - 1];
  final result = data1 + data2;

  return int.parse(result);
}

void main() {
  final data = File('data.txt');
  num sum;
  List list = [];

  Stream<String> lines = data.openRead().transform(utf8.decoder).transform(
        LineSplitter(),
      );

  lines.forEach((line) {
    final result = trebuchetCombine(
      line.replaceAll(RegExp(r'[^0-9]'), ''),
    );
    list.add(result);
  }).then(
    (value) => {
      sum = list.reduce((a, b) => a + b),
      print('🥞 The result is $sum'),
    },
  );
}
